### PR TITLE
removing the InGroup functionality

### PR DESCRIPTION
**Pull request by martinsawicki**
due to excessive complexity and maintenance requirements relative to user value

### CODE REVIEW

File: azure/src/main/java/com/microsoft/azure/implementation/Azure.java [hunk 1]
```diff
@@ -211,11 +210,6 @@ public Azure withDefaultSubscription() throws CloudException, IOException {
     }
 
     public interface ResourceGroup extends com.microsoft.azure.management.resources.ResourceGroup {
```
Comment (jianghaolu) [inline azure/src/main/java/com/microsoft/azure/implementation/Azure.java]: This can be deleted. But we can do that in a separate pr.

File: azure/src/main/java/com/microsoft/azure/implementation/AzureResourceGroupImpl.java [hunk 2]
```diff
@@ -1,55 +1,14 @@
 package com.microsoft.azure.implementation;
```
Comment (jianghaolu) [inline azure/src/main/java/com/microsoft/azure/implementation/AzureResourceGroupImpl.java]: Same.

File: azure/src/test/java/com/microsoft/azure/AzureTests.java [hunk 3]
```diff
@@ -76,37 +72,6 @@ public void setup() throws Exception {
     }
 
     /**
-     * Tests the public IP address implementation from an individual service client
-     * @throws Exception 
-     */
-    @Test
-    public void testPublicIpAddressesInGroup() throws Exception {
```
Comment (jianghaolu) [inline azure/src/test/java/com/microsoft/azure/AzureTests.java]: Why deleting this test?

File: azure/src/main/java/com/microsoft/azure/implementation/Azure.java (hunk 1 above)
Comment (martinsawicki) [inline azure/src/main/java/com/microsoft/azure/implementation/Azure.java]: yes, I see what you mean. there may be some "empty shell" definitions I may have left behind.

File: azure/src/test/java/com/microsoft/azure/AzureTests.java (hunk 3 above)
Comment (martinsawicki) [inline azure/src/test/java/com/microsoft/azure/AzureTests.java]: note this is the InGroup test for PIPs

Comment (jianghaolu) [inline azure/src/test/java/com/microsoft/azure/AzureTests.java]: Deleting it would make it hard to track.
Replace with this?
```java
    @Test
    public void testPublicIpAddressesInGroup() throws Exception {
        final String suffix = String.valueOf(System.currentTimeMillis());
        final String newGroupName = "group" + suffix;
        final String newPipName = "pip" + suffix;
        ResourceGroup group = azure2.resourceGroups().define(newGroupName)
                .withRegion(Region.US_WEST)
                .create();

        PublicIpAddresses pips = azure2.publicIpAddresses();
        
        PublicIpAddress pip = pips.define(newPipName)
            .withRegion(Region.US_WEST)
            .withExistingGroup(group)
            .withDynamicIp()
            .withLeafDomainLabel(newPipName)
            .create();
        
        System.out.println("Public IP addresses count: " + pips.list(group.name()).size());
        
        pip.update()
            .withLeafDomainLabel(newPipName + "x")
            .apply();
        
        pips.delete(group.name(), pip.name());
        azure2.resourceGroups().delete(group.name());
    }
```

Comment (martinsawicki) [inline azure/src/test/java/com/microsoft/azure/AzureTests.java]: Note that we do have a proper, full blown e2e test for PIPs - see TestPublicIpAddress -- like the others I've written.
This was just an early test for the InGroup specifically.

Comment (jianghaolu) [inline azure/src/test/java/com/microsoft/azure/AzureTests.java]: I see. Didn't see that.
Can we move them into seperate packages?
No particular reason, just feel like the roll-up client should just be a wrapper on top of tested robust individual clients running underneath.
When a test fails we also want to catch it earlier in building the individual client packages instead of finding it in a later phase like the roll-up package.

Comment (martinsawicki) [inline azure/src/test/java/com/microsoft/azure/AzureTests.java]: sounds reasonable. But this PR hasn't changed the current situation. Let's look at this soon.
